analyzers/src/SonarAnalyzer.CSharp/Rules/UseStringCreate.cs

```diff
@@ -37,7 +37,11 @@ public sealed class UseStringCreate : SonarDiagnosticAnalyzer
     protected override void Initialize(SonarAnalysisContext context) =>
         context.RegisterCompilationStartAction(c =>
         {
-            if (!CompilationTargetsValidNetVersion(c.Compilation))
+            // The string.Create method with IFormatProvider parameter is only available from .NET 6.0
+            if (!c.Compilation.IsMemberAvailable(
+                KnownType.System_String,
+                "Create",
+                symbol => ((IMethodSymbol)symbol).Parameters.Any(x => KnownType.System_IFormatProvider.Matches(x.Type))))
```

Comment:
A bit more type-safe.
```suggestion
                symbol => symbol is IMethodSymbol { Parameters: { Length: >0 } parameters } && parameters.Any(x => KnownType.System_IFormatProvider.Matches(x.Type)))
```

An alternative would be
```cs
c.Compilation.IsMemberAvailable<IMethodSymbol>(
    KnownType.System_String,
    "Create",
    methodSymbol => methodSymbol.Parameters.Any(x => KnownType.System_IFormatProvider.Matches(x.Type)));

public static bool IsMemberAvailable<TMemberType>(this Compilation compilation, KnownType type, string memberName, Func<TMemberType, bool> memberCheck = null) where TMemberType: ISymbol
{
    //..
    var memberSymbols = containingType.GetMembers(memberName).OfType<TMemberType>();
    return memberCheck == null
            ? memberSymbols.Any()
            : memberSymbols.Any(memberCheck)
}
```

Reply:
Consider creating a `KnownMethods` extension method as suggested here:
https://github.com/SonarSource/sonar-dotnet/pull/8589#discussion_r1464642287

Reply:
Nice. Much easier to use.
As for the KnownMethods extensions I'd rather not use it. I doubt it would be referenced from anywhere else other than this rule. And I'd also run into trouble again with code coverage.